src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -236,69 +258,83 @@ struct ArrayForEach {
   ///         sequential (row-major) order.
   ///         Note: The lowest ArrayView.rank() must be greater than or equal
   ///         to the highest dim in ItrDims. TODO: static checking for this.
-  template <typename... ArrayViews, typename Mask, typename Function>
+  template <typename ArrayViewTuple, typename Mask, typename Function,
+            detail::IsTuple<ArrayViewTuple> = nullptr>
```

Comment:
[not a change request, just looking to learn if possible] If I correctly understand that the intention of this line is to sanity-check that `ArrayViewTuple` is indeed a tuple (I don't see an alternative implementation that is turned on when it's not a tuple, but maybe I just missed it?) ... then would it also work to delete `detail::IsTuple` and replace this last template parameter with a `static_assert(IsTupleImpl<ArrayViewTuple>::value);` in the function body? To me that's easier to understand and has fewer lines of code, but is there a reason to chose one over the other besides aesthetics?